### ISSUE TITLE
Any `CustomErrorException`s thrown within custom code are automatically converted to `WebApplicationException` which discards all error objects

### ISSUE THREAD

**Issue by lewimuchiri** [yahoo/elide#2912]
Whenever I throw a `CustomErrorException` within my lifecycle hooks, the exception gets converted to a `WebApplicationException` and all ErrorObjects are discarded. By the time the application flow reaches the custom ErrorMapper, the exception is already converted and the error objects are all lost.

I have the following dependencies:
```kotlin
implementation("com.yahoo.elide:elide-spring-boot-starter:7.0.0-pr2") {
    exclude(group = "org.springframework.boot", module = "spring-boot-starter-tomcat")
}
implementation("org.springframework.boot:spring-boot-starter-undertow")
implementation("org.glassfish.jersey.core:jersey-common:3.1.1")
```

_Sidenote_: I had to add `jersey-common` so that I could get a JAX-RS provider. Otherwise I was always getting an exception when throwing errors, that no JAX-RS Provider was found

## Expected Behavior
ErrorObjects should all be retained when we get to the error mapper

**Comment by lewimuchiri** [comment]
Just followed up and seen that the code that's causing this to happen is this one:
<img width="596" alt="image" src="https://user-images.githubusercontent.com/1880626/222166224-0f87e780-13fd-4c1f-aaa2-188e139703b5.png">

It's inside `com.yahoo.elide.graphql.QueryRunner`. It's throwing a new exception

**Comment by aklish** [comment]
I see the first issue with Jersey, but not the second when I throw a CustomErrorException from a lifecycle hook.  My custom error hits the error mapper and it includes all the error objects.  Are you sure your lifecycle hook is triggering the exception (and not some other error)?

```
HTTP/1.1 200 OK
Date: Sun, 05 Mar 2023 01:27:18 GMT
Vary: Origin
Vary: Access-Control-Request-Method
Vary: Access-Control-Request-Headers
X-Content-Type-Options: nosniff
X-XSS-Protection: 0
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: 0
X-Frame-Options: SAMEORIGIN
Content-Type: application/json
Content-Length: 47

{
    "errors": [
        {
            "message": "Custom Error Message"
        }
    ]
}
```

**Comment by aklish** [comment]
Ok - I see the second issue now too for PREFLUSH and PRESECURITY hooks.

**Comment by aklish** [comment]
So it looks like graphql-java swallows the exception:
![Screen Shot 2023-03-05 at 3 07 39 PM](https://user-images.githubusercontent.com/6979393/222986085-6d33b899-76e8-4ddd-95e7-766a5fe74baf.png)

The exception is there, but there is no method on `GraphQLError` to get at it.  In short, we can fix the first problem (the jersey issue), but I don't know of a solution to fix the 2nd without a corresponding patch to graphql-java.

**Comment by lewimuchiri** [comment]
Yes, correct. I'm throwing my exception in a PRESECURITY hook. 
For the 'ErrorObjects gettign lost' problem, I suppose we can resolve this by defining a custom exception that implements `GraphQLError`. We can then add our extra errors to the `extensions` block in the Graphql error

**Comment by lewimuchiri** [comment]
Again, using Elide, there's no way of overriding the `DataFetcherExceptionHandler` from Graphql so that you can throw your own custom exceptions as you'd like. This is because we have to do something like this:
```java
@Bean
public GraphQL graphQL(GraphQLSchema schema) {
    return GraphQL.newGraphQL(schema)
            .queryExecutionStrategy(new AsyncExecutionStrategy(new CustomDataFetcherExceptionHandler()))
            .mutationExecutionStrategy(new AsyncSerialExecutionStrategy(new CustomDataFetcherExceptionHandler()))
            .build();
}
```
which gives the error `No beans of GraphQLSchema type found`

**Comment by aklish** [comment]
We could probably find a way to expose that so you can customize the errors.  We currently have a bean GraphQLConfiguration - but it is not that useful for something like this.

**Comment by lewimuchiri** [comment]
When will fix this get to Elide version 7? I'm still using version `7.0.0-pr2` (due to Spring Boot 3 support)

**Comment by aklish** [comment]
Released yesterday in 7.0.0-pr3

**Comment by lewimuchiri** [comment]
Thanks. Good stuff
